### PR TITLE
fix(handle-errors/Dockerfile)

### LINKED ISSUE #2: make fails during the docker build phase of handle-errors using Go v1.16
Running `make` results in a build failure, using
Go: `go version go1.16.2 linux/amd64`
```
49a2b4b8f08b8bb57b6e826a4888dff36a607e33b8a779a76b06d534b3897acb
Created network traefik
Building docker images...
redis uses an image, skipping
whoami uses an image, skipping
docker-socket-proxy uses an image, skipping
openldap uses an image, skipping
postgres uses an image, skipping
authelia uses an image, skipping
traefik uses an image, skipping
Building handle-errors
[+] Building 38.1s (10/14)
 => [internal] load build definition from Dockerfile                                   0.1s
 => => transferring dockerfile: 662B                                                   0.0s
 => [internal] load .dockerignore                                                      0.1s
 => => transferring context: 2B                                                        0.0s
 => [internal] load metadata for docker.io/library/golang:alpine                       3.3s
 => [auth] library/golang:pull token for registry-1.docker.io                          0.0s
 => [builder 1/5] FROM docker.io/library/golang:alpine@sha256:3411aef9ae9cb0fe3534fe  31.8s
 => => resolve docker.io/library/golang:alpine@sha256:3411aef9ae9cb0fe3534fe2a4d1a974  0.0s
 => => sha256:12d5f94cd4d2840e538e82e26a5dfddf711b30cc98a9f6e01bcf65d 1.36kB / 1.36kB  0.0s
 => => sha256:19b59f0222410f71faae89932a13c75635f147de093c7a014931f50 5.18kB / 5.18kB  0.0s
 => => sha256:448433d692de67fadc3e270369294f10bbc32683e28e4144e7d 281.27kB / 281.27kB  0.3s
 => => sha256:7c2a3d42746fcfc7f036d78c91f23a708eccc332efd705161238e6aafc5 153B / 153B  0.2s
 => => sha256:f6d283d788a6f5f59325116bda7d61102a1bc7f74d902d617e 105.69MB / 105.69MB  21.7s
 => => sha256:3411aef9ae9cb0fe3534fe2a4d1a9745d952d9a5ed1e20a11ff1054 1.65kB / 1.65kB  0.0s
 => => sha256:757f90a0dc844e1ca0ce652556adf5fddaaab3b13d949b52fc384525af8 156B / 156B  0.4s
 => => extracting sha256:448433d692de67fadc3e270369294f10bbc32683e28e4144e7d2d2fedbf6  0.8s
 => => extracting sha256:7c2a3d42746fcfc7f036d78c91f23a708eccc332efd705161238e6aafc55  0.0s
 => => extracting sha256:f6d283d788a6f5f59325116bda7d61102a1bc7f74d902d617ed62cf90c58  9.3s
 => => extracting sha256:757f90a0dc844e1ca0ce652556adf5fddaaab3b13d949b52fc384525af8c  0.0s
 => [internal] load build context                                                      0.1s
 => => transferring context: 15.21kB                                                   0.0s
 => [builder 2/5] RUN adduser     --disabled-password     --gecos ""     --home "/non  2.1s
 => [builder 3/5] WORKDIR /go/src                                                      0.1s
 => [builder 4/5] COPY main.go ./                                                      0.1s
 => ERROR [builder 5/5] RUN CGO_ENABLED=0 go build     -ldflags='-w -s -extldflags "-  0.4s
------
 > [builder 5/5] RUN CGO_ENABLED=0 go build     -ldflags='-w -s -extldflags "-static"' -a     -o /go/bin/serve .:
#10 0.415 go: go.mod file not found in current directory or any parent directory; see 'go help modules'
------
executor failed running [/bin/sh -c CGO_ENABLED=0 go build     -ldflags='-w -s -extldflags "-static"' -a     -o /go/bin/serve .]: exit code: 1
ERROR: Service 'handle-errors' failed to build
make: *** [Makefile:14: .build] Error 1
```

Go as of `v1.11` appears to understand and by default require a `go.mod` file describing the module being built to eliminate the need of the `$GOPATH`.  As a non-expert in Go, it appears between `v1.11` and `v1.16`, `$GO111MODULE` could be used to circumvent the default expectation; however, going forward with `v1.17` it seems like that will no longer be an option.

[New module changes in Go 1.16](https://blog.golang.org/go116-module-changes)

Perhaps addressing this would suggest the need for a separate repository for the error handler.  However, for purposes of demonstration, even if not best practice, simply adding a basic module definition file and referencing in the necessary build step of the `handle-errors/Dockerfile` would be sufficient.  In my local environment that seems to be enough to allow the project to again build with my tool-chain and nominally exercise the expected features.  I'll offer a pull request shortly.

### PR DESCRIPTION
**Pull request by jlcummings**
Fixes #2 

fulfill the golang v1.11+ requirement to use `go.mod`

for golang v1.11+ a `go.mod` file is required for each buildable module, that includes the name (or namespace) of the base (or parent) module being built, dependencies, and golang version that it is being built under; a workaround exists for golang versions between v1.11 and v1.16 inclusively, but will not be available in v1.17 and beyond, whereby you can exercise the $GO111MODULE environment variable.

BREAKING CHANGE: the go module system was introduced in v1.11, and so this may likely not be a backwards compatible change

### CODE REVIEW

File: handle-errors/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ RUN adduser \
     "${USER}"
 
 WORKDIR $GOPATH/src
+COPY go.mod ./
 COPY main.go ./
```
Comment (jamescurtin) [inline handle-errors/Dockerfile:17]: Style nit: the new file could be added to the existing COPY layer.

```suggestion
COPY main.go go.mod ./
```

File: handle-errors/go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module jamescurtin/handle-errors
+
+go 1.16
```
Comment (jamescurtin) [inline handle-errors/go.mod:3]: What do you think about updating the golang image tag in `handle-errors/Dockerfile` such that it is pinned to a 1.16 specifically? That would help us avoid potential situations in the future were there are breaking changes that go un-noticed.

Switching from `golang:alpine` to `golang:1.16-alpine3.13` should do the trick

**Review by jamescurtin — APPROVED** [review]
Looks great! Two small comments below

File: handle-errors/go.mod (hunk 2 above)
Comment (jlcummings) [reply handle-errors/go.mod:3]: Definitely the way to go.  I’ll include it later tonight.

**Review by jlcummings — COMMENTED** [review]
LGTM

File: handle-errors/go.mod (hunk 2 above)
Comment (jlcummings) [reply handle-errors/go.mod:3]: Amending this slightly, so the import path reflects the actual repository here on github.

**Review by jamescurtin — APPROVED** [review]

**Pull request merged**